kythe/docs/asciidoc.bzl

```diff
@@ -11,38 +17,51 @@ def _asciidoc_impl(ctx):
     if ctx.attr.example_script:
         args += ["--attribute=example_script=" + ctx.file.example_script.path]
     args += ["--conf-file=%s" % c.path for c in ctx.files.confs]
-    args += ["-o", ctx.outputs.out.path]
+    args += ["-o", "out/" + ctx.attr.name + ".html"]
     args += [ctx.file.src.path]
 
     # Get the path where all our necessary tools are located so it can be set
     # to PATH in our run_shell command.
-    tool_path = ctx.toolchains["//tools/build_rules/external_tools:external_tools_toolchain_type"].path
+    tool_path = ctx.toolchains[_toolchain_type].path
 
+    # Declare the logfile as an output so that it can be read if something goes
+    # awry (otherwise Bazel will clean it up).
     logfile = ctx.actions.declare_file(ctx.attr.name + ".logfile")
 
-    # Run asciidoc, capture stderr, look in stderr for error messages and fail if we find any.
+    # Run asciidoc and capture stderr to logfile. If it succeeds, look in the
+    # captured log for error messages and fail if we find any.
     ctx.actions.run_shell(
-        inputs = [ctx.file.src] + ctx.files.confs + ([ctx.file.example_script] if ctx.file.example_script else []) + ctx.files.data,
+        inputs = ([ctx.file.src] +
+                  ctx.files.confs +
+                  ([ctx.file.example_script] if ctx.file.example_script else []) +
+                  ctx.files.data),
         outputs = [ctx.outputs.out, logfile],
-        env = {
-            "PATH": tool_path,
-        },
+        env = {"PATH": tool_path},  # so we can locate the binaries asciidoc needs
         command = "\n".join([
-            '%s %s 2> %s' % (
+            # Create the temporary staging directory.
+            "mkdir out",
```

Comment:
use https://docs.bazel.build/versions/master/skylark/lib/actions.html#declare_directory and provide the path to that declared directory here rather than hardcoding `out`?

Reply:
I did that originally, but there wasn't any point to it—the directory is a nonce, and doesn't outlive the rule, so declaring it doesn't add any value (and complects the code quite a bit)